src/kakarot/instructions/system_operations.cairo

```diff
@@ -476,58 +409,77 @@ namespace CallHelper {
     // @param calling_ctx The pointer to the calling execution context.
     // @param with_value The boolean that determines whether the sub-context's calling context has a value read from the calling context's stack or the calling context's calling context.
     // @param read_only The boolean that determines whether state modifications can be executed from the sub-execution context.
+    // @param self_call A boolean to indicate whether the CallContext is self (current address) or the args address
```

Comment:
i think the comment is not so clear we could add a bit more to explain that `self_call` encapsulates the logic of separating an address to get code and an address call into
of CALL, STATICCALL vs. CALLCODE and DELEGATE CALL 

maybe:
```
// A boolean to indicate whether the account to message-call into is self (address of the current executing account) or the call argument's address (address of the call's target account)

Reply:
Could we maybe use the naming of the execution specs, namily [is_staticcall](https://github.com/ethereum/execution-specs/blob/master/src/ethereum/shanghai/vm/instructions/system.py#L274).

Reply:
I'm ok with the comment from @Eikix, not sure about what you mean @greged93 since it's not about static or not, but about, as elias said, the account updated by the execution